routes/status.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+use Illuminate\Support\Facades\Route;
+use Illuminate\Support\Facades\DB;
+use Symfony\Component\HttpFoundation\Response;
+
+/*
+|--------------------------------------------------------------------------
+| Web Routes
+|--------------------------------------------------------------------------
+|
+| Here is where you can register web routes for your application. These
+| routes are loaded by the RouteServiceProvider within a group which
+| contains the "web" middleware group. Now create something great!
+|
+*/
+
+// Database connection check
+Route::get('/db', function() {
+    try {
+        DB::connection()->getPdo();
+        return Response("Connection successful", 200)
+        ->header('Content-Type', 'text/plain');
+    } catch (\Exception $e) {
+        return Response("Error connecting to database.", 500)
+        ->header('Content-Type', 'text/plain');
+    }
+});
+
+// plain web check NGINX & PHP working
+Route::get('/', function() {
+    return Response("Connection successful", 200)
+    ->header('Content-Type', 'text/plain');
+});
```

Comment:
I think this will get confusing fast if issues happen. You need to remember that one healthcheck is only a no-op ping that only says the app not stuck and the other is actually a DB connection only (even this one won't alert if the database is gone, maybe it can be improved with an actual lightweight query that only retrieves a single record or only describes some important table, etc..).

Additionally, which one do you plan to use as the liveness check and which one as the readiness check? The DB or the no-op ping?

If we introduce Redis or some other component like an API that the app really depends on, are we adding other `/status/$path` endpoints?

Reply:
At this point we only intend to use `/status/db` for both readiness and liveness probes.

The other routes are if we decide to separate the 2 in the future.

As far as increasing the depth of the database checks I don't see the need for an internal K8 check for the application pod. If we are concerned about database health, so that the MySQL pod can be restarted I would probably do that more direct to the MySQL pod rather than going through the application pod.

We will also have external probes that will be checking full page rendering.